### ISSUE TITLE
[apigateway] support Stage attribute as part of the BasePathMapping resource

### ISSUE THREAD

**Issue by liukkonend** [aws/aws-cdk#4857]
There is no ability to explicitly specify a deployment stage when creating a custom domain. It seems to take the most recent deployment (if available).

### Reproduction Steps
```typescript
export class BasePathMapping extends Resource {
  constructor(scope: Construct, id: string, props: BasePathMappingProps) {
    super(scope, id);

    if (props.basePath && !Token.isUnresolved(props.basePath)) {
      if (!props.basePath.match(/^[a-z0-9$_.+!*'()-]+$/)) {
        throw new Error(`A base path may only contain letters, numbers, and one of "$-_.+!*'()", received: ${props.basePath}`);
      }
    }

    // if this is an owned API and it has a deployment stage, map all requests
    // to that stage. otherwise, the stage will have to be specified in the URL.
    const stage = props.restApi instanceof RestApi
      ? props.restApi.deploymentStage
      : undefined;

    new CfnBasePathMapping(this, 'Resource', {
      basePath: props.basePath,
      domainName: props.domainName.domainName,
      restApiId: props.restApi.restApiId,
      stage: stage && stage.stageName,
    });
  }
}
```

### Environment

  - **CLI Version      :** 1.13.1
  - **OS               :** MacOS
  - **Language         :** TypeScript

--- 

This is :bug: Bug Report

**Comment by jeremypearson** [comment]
I don't think this is related to your issue, as you don't seem to be using the SAM CfnApi.

But just in case anyone comes across this issue when searching for how to set the `stage` property of a CfnBasePathMapping (which is imported from '@aws-cdk/aws-apigateway') to the `Stage` variable of a CfnApi (which it imported from '@aws-cdk/aws-sam'), it seems like I've been able to set it like so:
```ts
stage: { Ref: `${myApi.logicalId}.Stage` }
```

**Comment by Litee** [comment]
I think this problem is resolved now, in the latest CDK (currently v1.115) there is a `stage` parameter.

**Comment by nija-at** [comment]
Indeed it was fixed here - https://github.com/aws/aws-cdk/commit/bfc4b5c6b050f9541dfc77999cb043f2f8d03626

Thanks @Litee
